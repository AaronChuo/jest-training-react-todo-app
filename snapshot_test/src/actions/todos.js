import {
  ADD_TODO,
  DELETE_TODO,
  EDIT_TODO,
  COMPLETE_TODO,
  COMPLETE_ALL,
  CLEAR_COMPLETED,
} from 'constants/todos';

export const addTodo = (payload) => {
  return {
    type: ADD_TODO,
    payload,
  };
};

export const deleteTodo = (payload) => {
  return {
    type: DELETE_TODO,
    payload,
  };
};

export const editTodo = (payload) => {
  return {
    type: EDIT_TODO,
    payload,
  };
};

export const completeTodo = (payload) => {
  return {
    type: COMPLETE_TODO,
    payload,
  };
};

export const completeAll = () => {
  return {
    type: COMPLETE_ALL,
  };
};

export const clearCompleted = () => {
  return {
    type: CLEAR_COMPLETED,
  };
};
