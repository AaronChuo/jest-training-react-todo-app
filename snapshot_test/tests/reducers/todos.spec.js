import * as actions from 'actions/todos';
import todosReducer from 'reducers/todos';
import { todoList } from 'mock-data';

describe('Todos reducer', () => {
  it('should return initial state', () => {
    expect(todosReducer(undefined, {})).toMatchSnapshot();
  });

  it('should add new todo', () => {
    const action = actions.addTodo({
      text: 'meeting with team members',
    });
    expect(todosReducer(todoList, action)).toMatchSnapshot();
  });

  it('should delete the todo', () => {
    const action = actions.deleteTodo({
      id: 1,
    });
    expect(todosReducer(todoList, action)).toMatchSnapshot();
  });

  it('should edit the todo', () => {
    const action = actions.editTodo({
      id: 1,
      text: 'dinner with boss',
    });
    expect(todosReducer(todoList, action)).toMatchSnapshot();
  });

  it('should complete the todo', () => {
    const action = actions.completeTodo({
      id: 1,
    });
    expect(todosReducer(todoList, action)).toMatchSnapshot();
  });

  it('should complete all todos', () => {
    const action = actions.completeAll();
    expect(todosReducer(todoList, action)).toMatchSnapshot();
  });

  it('should clear completed todo', () => {
    const action = actions.clearCompleted();
    expect(todosReducer(todoList, action)).toMatchSnapshot();
  });

  it('should return current state with wrong action', () => {
    const action = {
      type: 'WRONG_ACTION',
    };
    expect(todosReducer(todoList, action)).toMatchSnapshot();
  });
});
