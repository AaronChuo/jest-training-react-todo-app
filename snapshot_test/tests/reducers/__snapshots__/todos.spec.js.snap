// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos reducer should add new todo 1`] = `
Array [
  Object {
    "completed": false,
    "id": 2,
    "text": "meeting with team members",
  },
  Object {
    "completed": false,
    "id": 0,
    "text": "Use Redux",
  },
  Object {
    "completed": true,
    "id": 1,
    "text": "Use CodeSandbox",
  },
]
`;

exports[`Todos reducer should clear completed todo 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Use Redux",
  },
]
`;

exports[`Todos reducer should complete all todos 1`] = `
Array [
  Object {
    "completed": true,
    "id": 0,
    "text": "Use Redux",
  },
  Object {
    "completed": true,
    "id": 1,
    "text": "Use CodeSandbox",
  },
]
`;

exports[`Todos reducer should complete the todo 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Use Redux",
  },
  Object {
    "completed": false,
    "id": 1,
    "text": "Use CodeSandbox",
  },
]
`;

exports[`Todos reducer should delete the todo 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Use Redux",
  },
]
`;

exports[`Todos reducer should edit the todo 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Use Redux",
  },
  Object {
    "completed": true,
    "id": 1,
    "text": "dinner with boss",
  },
]
`;

exports[`Todos reducer should return current state with wrong action 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Use Redux",
  },
  Object {
    "completed": true,
    "id": 1,
    "text": "Use CodeSandbox",
  },
]
`;

exports[`Todos reducer should return initial state 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Use Redux",
  },
  Object {
    "completed": true,
    "id": 1,
    "text": "Use CodeSandbox",
  },
]
`;
